src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -23,4 +23,20 @@
     public boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
         return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
     }
+
+    /**
+     * Returns true if more than one {@code Prefix} is present in the given
+     * {@code ArgumentMultimap}.
+     */
+    public boolean areMoreThanOnePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).filter(prefix -> argumentMultimap.getValue(prefix).isPresent()).count() > 1;
+    }
+
+    /**
+     * Returns true if no {@code Prefix} is present in the given
+     * {@code ArgumentMultimap}.
+     */
+    public boolean areNoPrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).filter(prefix -> argumentMultimap.getValue(prefix).isPresent()).count() == 0;
+    }
```

Comment:
Nice use of streams 🤯

Reply:
```suggestion
    public boolean areMoreThanOnePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
        return Stream.of(prefixes).filter(prefix -> argumentMultimap.getValue(prefix).isPresent()).count() > 1;
    }

    /**
     * Returns false if no {@code Prefix} is present in the given {@code ArgumentMultimap}.
     */
    public boolean isAnyPrefixPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
        return Stream.of(prefixes).filter(prefix -> argumentMultimap.getValue(prefix).isPresent()).count() != 0;
    }
```

i think this is a better name for the function haha

Reply:
I'm also confused by the need for the function on top, can't we just use the negated value from the second function?

Reply:
These are the current checks we have:
1. Checks if all prefixes are present (we use this to check if _exactly one_ particular prefix is present, by passing only that prefix in)
2. Checks if no prefixes are present (to catch errors for situations where we need at least one prefix)
3. Checks if more than one prefix are present. 

I think the difference between negated 2 and 3 is that negated 2 also accounts for _only one_ prefix being present, but 3 is meant to catch errors for situations where we cannot have _more _than__ one prefix present (but exactly one prefix is okay)

Are there any overlaps you see? I was also thinking about redundancies